pkg/controllers/routercerts/controller.go

```diff
@@ -207,6 +208,13 @@ func (c *routerCertsDomainValidationController) validateRouterCertificates() ope
 		return newRouterCertsDegradedf("InvalidServerCertRouterCerts", "secret/%v.spec.data[%v] -n %v: certificate could not validate route hostname %v: %v", c.defaultSecretName, ingressDomain, c.secretNamespace, verifyOptions.DNSName, err)
 	}
 
+	// check the server certificates for reliance on the deprecated CN field
+	for _, serverCert := range serverCerts {
+		if !crypto.CertHasSAN(serverCert) {
+			return newRouterCertsNoUpgradef("InvalidServerCertRouterCerts", "secret/%v.spec.data[%v] -n %v (%v): certificate relies on legacy Common Name field, use SANs instead", c.defaultSecretName, ingressDomain, c.secretNamespace, serverCert.SerialNumber)
+		}
+	}
+
 	// we made it this far without a problem
 	return operatorv1.OperatorCondition{
 		Type:   conditionRouterCertsDegradedType,
```

Comment:
i guess this has to be `conditionRouterCertsNoUpgradeType`

Reply:
OK that's going to be more complicated than that then. The `sync` loop currently assumes that we only operate one condition type in this function; operating `NoUpgrade` will require a little refactoring.

Reply:
/hold